core/src/main/java/com/google/googlejavaformat/java/Main.java

```diff
@@ -207,6 +207,7 @@ private int formatFiles(ArgInfo argInfo, JavaFormatterOptions options) {
     int numThreads = Math.min(MAX_THREADS, argInfo.parameters.fileNamesFlag.size());
     ExecutorService executorService = Executors.newFixedThreadPool(numThreads);
 
+    Map<Path, String> inputs = new LinkedHashMap<>();
```

Comment:
One last thought: what about changing the existing map collecting `FormatResult` instead of `Future<String>`? With `FormatResult` having a pointer to the input and the callable?